solidity/contracts/peripherals/SecurityPool.sol

```diff
@@ -0,0 +1,374 @@
+
+	PriceOracleManagerAndOperatorQueuer public priceOracleManagerAndOperatorQueuer;
+	OpenOracle public openOracle;
+
+	event SecurityBondAllowanceChange(address vault, uint256 from, uint256 to);
+	event PerformWithdrawRep(address vault, uint256 amount);
+	event PoolRetentionRateChanged(uint256 retentionRate);
+	event ForkSecurityPool(uint256 repAtFork);
+	event MigrateVault(address vault, QuestionOutcome outcome, uint256 poolOwnership, uint256 securityBondAllowance);
+	event TruthAuctionStarted(uint256 completeSetCollateralAmount, uint256 repMigrated, uint256 repAtFork);
+	event TruthAuctionFinalized();
+	event ClaimAuctionProceeds(address vault, uint256 amount, uint256 poolOwnershipAmount, uint256 poolOwnershipDenominator);
+	event MigrateRepFromParent(address vault, uint256 parentSecurityBondAllowance, uint256 parentpoolOwnership);
+	event DepositRep(address vault, uint256 repAmount, uint256 poolOwnership);
+
+	modifier isOperational {
+		(,, uint256 forkTime) = zoltar.universes(universeId);
+		require(forkTime == 0, 'Zoltar has forked');
+		require(systemState == SystemState.Operational, 'System is not operational');
+		_;
+	}
+
+	constructor(ISecurityPoolFactory _securityPoolFactory, OpenOracle _openOracle, ISecurityPool _parent, Zoltar _zoltar, uint192 _universeId, uint56 _questionId, uint256 _securityMultiplier) {
+		universeId = _universeId;
+		securityPoolFactory = _securityPoolFactory;
+		questionId = _questionId;
+		securityMultiplier = _securityMultiplier;
+		zoltar = _zoltar;
+		parent = _parent;
+		openOracle = _openOracle;
+		if (address(parent) == address(0x0)) { // origin universe never does truthAuction
+			systemState = SystemState.Operational;
+		} else {
+			systemState = SystemState.ForkMigration;
+			truthAuction = new Auction{ salt: bytes32(uint256(0x1)) }(address(this));
+		}
+		// todo, we can probably do these smarter so that we don't need migration
+		completeSet = new CompleteSet{ salt: bytes32(uint256(0x1)) }(address(this));
+	}
+
+	function setStartingParams(uint256 _currentRetentionRate, uint256 _repEthPrice, uint256 _completeSetCollateralAmount) public {
+		require(msg.sender == address(securityPoolFactory), 'only callable by securityPoolFactory');
+		lastUpdatedFeeAccumulator = block.timestamp;
+		currentRetentionRate = _currentRetentionRate;
+		completeSetCollateralAmount = _completeSetCollateralAmount;
+		(repToken,,) = zoltar.universes(universeId);
+		priceOracleManagerAndOperatorQueuer = new PriceOracleManagerAndOperatorQueuer{ salt: bytes32(uint256(0x1)) }(openOracle, this, repToken);
+		priceOracleManagerAndOperatorQueuer.setRepEthPrice(_repEthPrice);
+	}
+
+	function updateCollateralAmount() public {
+		(uint64 endTime,,,) = zoltar.questions(questionId);
+		uint256 clampedCurrentTimestamp = block.timestamp > endTime ? endTime : block.timestamp;
+		uint256 clampedLastUpdatedFeeAccumulator = lastUpdatedFeeAccumulator > endTime ? endTime : lastUpdatedFeeAccumulator;
```

Comment:
Solidity doesn't have a min built-in function?

Since we are doing this in at least 2 places, it may be worth making a named function just for readability.  Does Solidity do inline optimizations for pure private functions like this, so such a thing ends up being "free" gas wise?